terraform/azure-cosmosdb/provision-cosmosdb-sql.tf

```diff
@@ -53,9 +55,9 @@ provider "azurerm" {
 }
 
 locals {
-  resource_group = length(var.resource_group) == 0 ? format("rg-%s", var.instance_name) : var.resource_group
-
+  resource_group                = length(var.resource_group) == 0 ? format("rg-%s", var.instance_name) : var.resource_group
   enable_virtual_network_filter = (var.authorized_network != "")
+  private_endpoint_enabled      = var.private_endpoint_subnet_id == null ? false : length(var.private_endpoint_subnet_id) > 0 ? true : false
```

Comment:
thoughts: you might want to simplify the logic to improve the readability

```
private_endpoint_enabled = var.private_endpoint_subnet_id == null || length(var.private_endpoint_subnet_id) == 0 ? false : true
```

Reply:
Hello, it fixed though I' m not sure to understand syntax as I can' t find a related example in Terraform documentation

Reply:
Thank you very much @jlogel90. No worries, the logic applied in ternary operators is usually quite tricky. The logic was correct, but it is easier to read after this simplification